libres/tests/analysis/ies/test_ies_enkf_main.cpp

```diff
@@ -46,43 +43,135 @@ TEST_CASE("ies_enkf_linalg_extract_active", "[analysis]") {
     ies_enkf_data_fclose_log(data);
 
     // Test that `ies_enkf_linalg_extract_active` does nothing when all observations and realizations are active
-    matrix_type *E = matrix_alloc(state_size, ens_size);
-    ies_enkf_linalg_extract_active(data, E, stdout, false);
-    REQUIRE(matrix_equal(Ein, E));
+    {
```

Comment:
So we need to talk about these braces! 😄 
I've heard about old-school variable-scoping in C, but is this necessary?

Reply:
> So we need to talk about these braces! smile

We'll find a good counsellor ... :-)

> I've heard about old-school variable-scoping in C, but is this necessary?

It is not necessary; and I am not taking the fight. But quite frankly I find it a bit amusing that everyone dislikes these scopes so much - in my opinion the use of scopes like is the absolute lowest level of creating small manageable units - not a full function/method - but more than just flat code. For me this structure:
```C++
{
   int local_var = ...;
   local_var = 100;
   ...
}
..
{
    int local_var = 200;
    ...
}
```
helps with two things:
1. It creates a very clear visual grouping of related statements.
2. It guarantees that both blocks have a pristine version of `local_var` - with no pollution between the blocks.

In some cases the correct thing would probably be to extract functions, but this is a poor mans extracted function.

Anyway - I'll fold any day.

Reply:
So these groups are quite uncommon. They come from a time where variables could only be declared at the beginning of a block, so it was a different time...

The norm for adding these blocks is that they would be there for a reason. So I often loose some time trying to figure out whether the additional curlies are doing anything to the code, before I figure out that they are just redundant. For instance, in C, a destructor for an object might be called early if you have inserted brackets. So its a bit of a surprise when they do nothing.

Reply:
Thanks for the examples.

Having mostly used languages without curly braces, I find they pollute more than they clear up.
I'm sure this is just a matter of habit so not really important.

As for your second example where `local_var` is re-initialised, to me this signals that we perhaps want a function instead.
But also here it's a matter of taste, so "smaken er som baken" and all that :)

Since most people, or at least most of the people I've discussed extra curly braces with don't like them, let's do the democratic thing and leave them behind.

Reply:
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Res-scope: 

**Alternative formulation:** Don’t declare a name in an unnecessarily large scope.

Reply:
I fully agree with the guidelines, and as far as I can tell they recommend splitting into several functions instead of creating scopes with curly braces.
Am I perhaps misunderstanding?

Reply:
Well - the message is "Don't declare a name in an unnecessarily large scope."; that can be achieved in several ways. Splitting into functions is certainly a way to achieve that, but it can also be achieved by defining variables in a lexical scope with `{ ... }`.

Anyway - I promise on the grave om childhood dog that this is the last nag I will make on the topic.